### DISCUSSION TITLE
Keyboard shortcuts when focus is not on home.

### DISCUSSION THREAD

**Discussion by mathieu-fournier** [iTwin/appui#804] [Q&A]
I am looking to add a keyboard shortcut using UIFramework : 
![image](https://github.com/iTwin/appui/assets/82905333/52b57bec-5d89-438b-a87b-533db863804d)

The issue I am facing is that the shortcut is only active when the focus is on home. 
When the focus is on something else, this is evaluated to false and my keyboard shortcut is ignored. 
![image](https://github.com/iTwin/appui/assets/82905333/a720f3f6-ace9-4146-8dad-e44f3fe6197c)

I want to have generic global keyboards inputs. 
I will control when they should be ignored or not. 
Is there a way to do that currently ? 

I found a workaround by overriding the `processShortcutKey` function of `FrameworkToolAdmin`. 
But this kinda defeats the whole purpose of having UIFramework handling keyboard shortcuts. 
![image](https://github.com/iTwin/appui/assets/82905333/7179df0c-6b10-49e4-858b-d3f01c349128)

thanks (;

**Comment by GerardasB** [comment]
Hey, you could either override `processShortcutKey` of a `FrameworkToolAdmin` or just process keys in a custom hook since as long as you call `UiFramework.keyboardShortcuts.processKey` keyboard shortcuts should be handled correctly: https://github.com/iTwin/appui/blob/cc44ece16bbac80b70813de4d4509f803b3adb6e/docs/storybook/src/keyboard/KeyboardShortcuts.tsx#L103

**Reply by mathieu-fournier** [reply]
This works, but it bypasses the whole UIFramework keyboard shortcut handling code. 
I would like to use the framework instead of bypassing it and rewriting the keyboard handling code myself. 
Something I have in mind that would do the trick is to add a props on KeyboardShortcutProps. 
Ex :
![image](https://github.com/iTwin/appui/assets/82905333/1df69230-f9b6-4b60-9de2-59cd01816b10)

**Reply by GerardasB** [reply]
I would really like to avoid going with this flag route, i.e. what if other applications will determine that they want to process shortcuts only if a specific element is focused (or only if some of the elements are not focused). Maybe instead we need a better logic on when the shortcuts should be handled?

Feel free to extend the `FrameworkToolAdmin` if you want to keep framework handling and only handle keyboard shortcuts of interest by combining `KeyboardShortcutContainer.generateKeyMapKey` and `UiFramework.keyboardShortcuts.getShortcut` (fallback to `FrameworkToolAdmin` for all other shortcuts)

**Reply by mathieu-fournier** [reply]
Opened a issue : 
https://github.com/iTwin/appui/issues/819
